### ISSUE TITLE
HTTPDirectResponse breaks envoy RDS if body is bigger than 4096bytes

### ISSUE THREAD

**Issue by netturpin** [istio/istio#42510]
### Bug Description

If the content of a body answer submitted as bytes in an [HTTPBody](https://istio.io/latest/docs/reference/config/networking/virtual-service/#HTTPBody) is bigger than 4096 bytes in an [HTTPDirectResponse](https://istio.io/latest/docs/reference/config/networking/virtual-service/#HTTPDirectResponse), it is rejected by envoy sidecars with the following message:
`scope":"envoy config","msg":"gRPC config for [type.googleapis.com/envoy.config.route.v3.RouteConfiguration](http://type.googleapis.com/envoy.config.route.v3.RouteConfiguration) rejected: response body size is 5764 bytes; maximum is 4096"}`. 

No other additional route will be injected in the envoy proxy. 
We found some explanation with the [envoy issue #8104](https://github.com/envoyproxy/envoy/issues/8104) that seems to confirm our hypothesis.

I would have expected either the payload to be able to be served and the envoy proxy to be kept in sync with istiod or the Virtualservice to be rejected by the validation webhook due to the size of the body. 

### Version

```prose
This occurred with the following versions:

istioctl version
client version: 1.16.1
control plane version: 1.15.3

kubectl version --short
Client Version: v1.25.5
Server Version: v1.24.5-gke.600
```

**Comment by hzxuzhonghu** [comment]
This can be tuned with envoyfilter. `MaxDirectResponseBodySizeBytes` has a default 4096 limit

**Comment by howardjohn** [comment]
You need to upgrade, its now hardcoded to 1mb.  https://github.com/istio/istio/commit/6f04fab1fa11342b6855b42fa85e549db06d9c76 (not sure that made it into a patch release yet)

**Comment by andrewhertog** [comment]
We are on 1.17.1, does that have this change in it?

**Comment by howardjohn** [comment]
Looks like it jsut missed the cut and will be in 1.17.2

**Comment by hzxuzhonghu** [comment]
Fixed by #44284, looks cherrypick to 1.17 failed https://github.com/istio/istio/issues/44290
